django/applications/catmaid/control/annotation.py

```diff
@@ -837,7 +836,7 @@ def _annotate_entities_with_name(project_id:Union[int,str], user_id, entity_ids)
         """.format(**{
             'n_list': '(' + '),('.join(map(lambda x: "{}, {}, {}, '{}'".format(
                     user_id, project_id, annotation_class.id, x), escaped_name_annotations)) + ')',
-        }))
+        })) # XXX is there an easier-to-read way to express this?
```

Comment:
Will remove this comment before merge

Reply:
Maybe this is more readable:

```
'n_list': ','.join(map(lambda x: f"({user_id}, {project_id}, {annotation_class.id}, '{x}')", escaped_name_annotations)) or '()',
```

Not by much though.

Reply:
How about splitting it over lines and using comprehensions instead of `map`/`lambda`?

```python
escaped_name_annotations = (n.replace("'", "''") for n in missing_name_annotations)
values = (f"({user_id}, {project_id}, {annotation_class.id}, {x})" for x in escaped_name_annotations)
values_str = ','.join(n_list) or '()'
cursor.execute("""
    INSERT INTO class_instance (user_id, project_id, class_id, name)
    VALUES {}
    RETURNING name, id;
""".format(values_str))
```

Using generator expressions avoids creating a bunch of intermediate lists.

Reply:
Whatever you agree on I'm happy to revise this towards.

Reply:
I like @clbarnes' solution better if the SQL becomes an f-string too for consistency.

Reply:
Good point! Edited the snippet.